.github/workflows/codeql.yml

```diff
@@ -0,0 +1,77 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main", hotfix-[0-9]*.[0-9]*.[0-9]*, release-* ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "main" ]
+  # At 11:27 PM, only on Monday, Wednesday, and Friday
+  schedule:
+    - cron: '27 23 * * 1,3,5'
```

Comment:
how are we going to use the scan report? Can't we use it for every PR like [this](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning-for-a-repository#understanding-the-pull-request-checks)?

Reply:
Is there any significance of the `27`? Can we make it `0`? 
(Just for OCD 😅)

Reply:
Regarding @adamxchen's comment, we can make it a required check for merging a PR once this change is merged in

Reply:
This one already runs on all PRs along with the schedule. The reason for the schedule is that vulnerabilities might get marked as such after a PR that pulled them in is merged.